01-ruby-fundamentals/07-oo-classes.md

```diff
@@ -63,50 +111,112 @@ Chair.new # => #<Chair:0x007fb7da550ca0> - this is an instance of a chair, it ha
 Chair.new.color # => "Black"
 ```
 
-## Every value is an object
-In the last section, we said *everything is an object*. This is a confusing concept at first, but let's go over a couple of examples to highlight that point.
-
-Every object has a class, try:
+This is a good start, our chairs now have something to do. But our chairs are still all the same. There's nothing to differentiate one chair from another.
 
 ```ruby
-"Hello".class # => String
-{}.class # => Hash
-Chair.new.class # => Chair
+first_chair = Chair.new
+second_chair = Chair.new
+
+first_chair.name     # => Terrible Chair
+second_chair.name    # => Terrible Chair
 ```
-__Question: What are the limitations of our `Chair` _class_?__
 
-### `initialize`
-Let's start over building our class, but let's build it to be a little customizable. We want to be able to customize the attributes `color` & `name` when we create our chair.
+### Adding State Through Instance Variables
+
+Ruby allows us to attach variables to an instance of a class. Such variables are appropriately named _instance variables_. To create or use an instance variable, prefix the variable's name with the `@` symbol. Instance variables can be read and written in any instance method.
+
+By default, instance variables are not accessible outside the instance they're attached to. By convention, to make them readable we define a method with the same name as the variable, which returns the current value of that variable. This method is called a _reader_, _getter_ or _accessor_, depending on what language you're in (Ruby uses _reader_).
```

Comment:
I wonder if it would be good to elaborate on this "not accessible outside" thing - or if it's ok if they're not quite bought into it yet. Not sure...